Add Lutron Serena tilt only wood blinds

wujiaqi opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Add support for Lutron (Caseta) Serena Wood Blinds, which are a tilt only blind. Currently all Lutron Caseta cover objects are handled as shades which is incorrect.

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->
The current implementation creates each device an object that only supports open/close as a SHADE type. Lutron Serena wood blinds are a tilt only blind so they should not use a plain open/close but instead TILT functions. This adds the additional type to support a device with tilt only operation. 
In addition this change maps the pylutron types to classes. I left comments for Sivoia QS Wood Blinds and RightDrawDrape as I do not own these products so I cannot test the implementation, but I don't expect they work correctly today anyhow.
Addresses the following issues:
- https://github.com/home-assistant/core/issues/65815
- https://github.com/home-assistant/core/issues/81524
- https://github.com/home-assistant/core/issues/81525
- https://github.com/home-assistant/core/issues/93679

- Link to documentation pull request: https://github.com/home-assistant/home-assistant.io/pull/31957

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] I have followed the [perfect PR recommendations][perfect-pr]
- [x] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [x] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [x] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [x] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [x] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [x] Untested files have been added to `.coveragerc`.

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

wujiaqi committed:
add support for Lutron Serena tilt only wood blinds

wujiaqi committed:
use set tilt 0 for close as lower shade seems to be inaccurate for blinds

bdraco commented on homeassistant/components/lutron_caseta/cover.py:
```diff
@@ -78,3 +64,93 @@ async def async_set_cover_position(self, **kwargs: Any) -> None:
+}
+
+
+def _map_covertype_pylutron_to_ha(pylutron_name):
+    try:
+        return PYLUTRON_TYPE_TO_CLASSES[pylutron_name]
+    except KeyError:
+        _LOGGER.error("%s cover type not implemented" | format(pylutron_name))
+        return None
+
```
This looks unused
```suggestion
```

bdraco commented on homeassistant/components/lutron_caseta/cover.py:
```diff
@@ -78,3 +64,93 @@ async def async_set_cover_position(self, **kwargs: Any) -> None:
+    "TriathlonRollerShade": LutronCasetaCover,
+    "QsWirelessShade": LutronCasetaCover,
+    "QsWirelessHorizontalSheerBlind": LutronCasetaCover,
+    # "QsWirelessWoodBlind": LutronCasetaCover, # implement tilt and lift wood blind
+    # "RightDrawDrape": LutronCasetaCover, # implement drape
```
Please remove commented code

bdraco commented on homeassistant/components/lutron_caseta/cover.py:
```diff
@@ -78,3 +64,93 @@ async def async_set_cover_position(self, **kwargs: Any) -> None:
+        """Return the current position of cover."""
+        return self._device["tilt"]
+
+    async def async_stop_cover_tilt(self, **kwargs: Any) -> None:
+        """Top the cover."""
+        await self._smartbridge.stop_cover(self.device_id)
```
There is some duplicate code here. Please refactor the `LutronCasetaCover` to have a base class so it can share so we don't have duplicates

wujiaqi replied:
actually i think this wasn't correct anyway, stop and tilt stop are separate methods

bdraco reviewed (dismissed):
Thanks @wujiaqi 

Please see review comments above

wujiaqi committed:
address cr comments

wujiaqi commented:
@bdraco thanks for reviewing, I addressed the comments

wujiaqi committed:
rename class to use more accurate term 'shade'

bdraco commented on homeassistant/components/lutron_caseta/cover.py:
```diff
@@ -78,3 +68,79 @@ async def async_set_cover_position(self, **kwargs: Any) -> None:
+    async def async_close_cover_tilt(self, **kwargs: Any) -> None:
+        """Close the cover."""
+        await self._smartbridge.set_tilt(self.device_id, 0)
+        await self.async_update()
```
Its unexpected to call `async_update` here... this may be an existing bug that was copied ..  will dig into this

bdraco replied:
Looks like it does update `self._device` in the base entity class.

Thats a bit messy but an existing issue

bdraco commented on homeassistant/components/lutron_caseta/cover.py:
```diff
@@ -78,3 +68,79 @@ async def async_set_cover_position(self, **kwargs: Any) -> None:
+
+    async def async_set_cover_tilt_position(self, **kwargs: Any) -> None:
+        """Move the shade to a specific position."""
+        if ATTR_TILT_POSITION in kwargs:
```
`ATTR_TILT_POSITION` is required so this can't be missing. Please remove the guard

https://github.com/home-assistant/core/blob/4da701a8e9ee880fbbfbaec24c237e1e6146a509/homeassistant/components/cover/__init__.py#L224

bdraco commented on homeassistant/components/lutron_caseta/cover.py:
```diff
@@ -78,3 +68,79 @@ async def async_set_cover_position(self, **kwargs: Any) -> None:
         if ATTR_POSITION in kwargs:
```
This guard is not needed since it can't be missing https://github.com/home-assistant/core/blob/4da701a8e9ee880fbbfbaec24c237e1e6146a509/homeassistant/components/cover/__init__.py#L181

This is an existing problem though

wujiaqi replied:
no problem I can clean it while I'm here

bdraco commented on homeassistant/components/lutron_caseta/cover.py:
```diff
@@ -17,27 +19,19 @@
 from .const import DOMAIN as CASETA_DOMAIN
 from .models import LutronCasetaData
 
+_LOGGER = logging.getLogger(__name__)
```
Looks unused now
```suggestion
```

bdraco commented on homeassistant/components/lutron_caseta/cover.py:
```diff
@@ -1,9 +1,11 @@
 """Support for Lutron Caseta shades."""
 
+import logging
```
```suggestion
```

wujiaqi committed:
address code review comments, removed base class as there are no common functions, removed stop tilt feature from blind as it does not work in pylutron

wujiaqi commented:
@bdraco ok I addressed your comments, except for the one regarding `async_update`, were you expecting any change for that one since it's existing?

I decided to remove the STOP_TILT feature from blinds, at first I thought it worked but maybe it was a figment of my imagination, when I tried testing it again looks like the capability doesn't work to stop a blind in the middle of an action in pylutron

bdraco commented:
> @bdraco ok I addressed your comments, except for the one regarding `async_update`, were you expecting any change for that one since it's existing?

All good to leave that as-is. Its an existing issue and it would be too much to address in this PR.
> 
> I decided to remove the STOP_TILT feature from blinds, at first I thought it worked but maybe it was a figment of my imagination, when I tried testing it again looks like using stop_cover to stop a blind in the middle of a tilt in pylutron doesn't work and I don't see any alternatives

👍

bdraco commented:
I need to find some time to test this when it won't be disruptive to my production system

bdraco commented:
I'm pushing this to my production now for testing

bdraco reviewed (approved):
Thanks @wujiaqi

bdraco commented:
Tested and working with my shades (I don't have tilt only so I was only looking for a regression)
